src/PostHog/Api/PostHogApiClientExtensions.cs

```diff
@@ -52,10 +52,11 @@ public static async Task<ApiResult> GroupIdentifyAsync(
         string type,
         StringOrValue<int> key,
         Dictionary<string, object>? properties,
-        CancellationToken cancellationToken)
+        CancellationToken cancellationToken,
+        string? distinctId = null)
```

Comment:
For public API's, I prefer to avoid optional parameters like this because it can cause [subtle versioning issues down the road](https://haacked.com/archive/2010/08/12/more-optional-versioning-fun.aspx/).

Instead, consider adding an overload with distinctId.